src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_controlnet.py

```diff
@@ -146,6 +331,11 @@ def __init__(
                 " checker. If you do not want to use the safety checker, you can pass `'safety_checker=None'` instead."
             )
 
+        if isinstance(controlnet, (list, tuple)):
+            controlnet = MultiControlNet(controlnet)
+        else:
+            controlnet = MultiControlNet([controlnet])
```

Comment:
```suggestion
            controlnet = controlnet
```

Think here sadly we have to keep the controlnet of type `ControlNetModel` because `MultiControlNet` can't be saved. 

So while this previously worked:

```py
controlnet = ControlNetModel.from_pretrained(
    "fusing/stable-diffusion-v1-5-controlnet-canny", torch_dtype=torch.float16
)

pipe = StableDiffusionControlNetPipeline.from_pretrained(
    "runwayml/stable-diffusion-v1-5", controlnet=controlnet, safety_checker=None, torch_dtype=torch.float16
)
pipe.save_pretrained("./controlnet_pipe")
pipe = StableDiffusionControlNetPipeline.from_pretrained("./controlnet_pipe")
```

it wouldn't work anymore now. Would be great if we could leave this is as is :-)

Reply:
Note that this means that "single" controlnet pipelines can be saved and loaded while multiple controlnet pipelines cannot. But I think this is ok for now as multi controlnet is more of an extension and saving could be handled later

Reply:
OK! understood.